packages/flutter/lib/src/rendering/mouse_tracker.dart

```diff
@@ -207,11 +207,16 @@ class MouseTracker extends ChangeNotifier {
       _debugDuringDeviceUpdate = true;
       return true;
     }());
-    task();
-    assert(() {
-      _debugDuringDeviceUpdate = false;
-      return true;
-    }());
+    try {
+      task();
+    } catch (e) {
+      rethrow;
```

Comment:
fly-by-comment: The catch clause can just be omitted, no?

Reply:
Catch is needed not to suppress the taken error. And, finally -- to reset the state independently of task-flow.

Reply:
you can do it without catch.
try {

} finally {

}

Reply:
Yeah, but in that case it'll be lost the original error: `Scaffold.geometryOf() must only be accessed during the paint phase`